lib/dao.js

```diff
@@ -1526,6 +1526,27 @@ function NumberType(val) {
   return !isNaN(num) ? num : val;
 }
 
+function coerceArray(val) {
+  if (Array.isArray(val)) {
+    return val;
+  }
+
+  if (!utils.isPlainObject(val)) {
+    throw new Error(g.f('Value is not an {{array}} or {{array-like object}}'));
+  }
+
+  var arrayVal = new Array(Object.keys(val).length);
```

Comment:
@bajtos Are you sure? I tried to implement this in a way that specifically avoids such an issue.

```
Scallop:Projects heath$ node -v
v6.9.1
Scallop:Projects heath$ node
> keys = Object.keys({ 4000000000: true })
[ '4000000000' ]
> a = new Array(keys.length)
[  ]
```